airflow/contrib/operators/gcs_to_gcs.py

```diff
@@ -44,12 +44,17 @@ class GoogleCloudStorageToGoogleCloudStorageOperator(BaseOperator):
         storage bucket.
         If a wildcard is supplied in the source_object argument, this is the
         prefix that will be prepended to the final destination objects' paths.
+        Note that the source path's part before the wildcard will be removed;
+        if it needs to be retained it should be appended to destionation_object.
     :type destination_object: string
     :param move_object: When move object is True, the object is moved instead
     of copied to the new location.
                         This is the equivalent of a mv command as opposed to a
                         cp command.
     :type move_object: bool
+    :param push_xcom: When push_xcom is True, the list of copied files is saved
```

Comment:
Can you give a scenario when this can be useful? i.e scenario where this list of copied filenames would be used by downstream tasks.

Also, do you think that adding source and destination bucket would be useful? or you can probably create a full path e.g `gs://{SOURCE_BUCKET}/{SOURCE_OBJECT}`.

Reply:
This is a moot point now, but for the discussion's sake, I don't think the buckets would be necessary -- whichever method was used to determine them for this task could be used in any subsequent ones.

The core use case is useful when using wildcards, as you can't know which files were copied until after you list and copy them, so there is no consistent way to reconstruct the list in a downstream task.